### PR TITLE
refactor: export `loadConfigFile` as a public function `LoadConfigFile`

### LINKED ISSUE #119056: Export `loadConfigFromFile` as a public function `LoadConfigFromFile`
### What would you like to be added?

I want `loadConfigFromFile` to be exported as a public function `LoadConfigFromFile` 
https://github.com/kubernetes/kubernetes/blob/746b88c6ff442185589640885228ef100811083a/cmd/kube-scheduler/app/options/configfile.go#L33-L40
 

### Why is this needed?

* [Kubernetes cluster-autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler) imports scheduler code and runs scheduler simulations for `Pending` pods to decide which nodegroup should be scaled up. 
* consider this case:
  1. a user is running a scheduler with custom scheduler config
  2. the user wants cluster-autoscaler to run scheduler simulations based on the custom scheduler config
 
For ii default scheduler config is used in cluster-autoscaler scheduler simulations. Right now, it is not possible to use custom scheduler config for cluster-autoscaler scheduler simulations  but I am working on a PR (https://github.com/kubernetes/autoscaler/pull/5708) to support this feature. For this feature, I need to parse scheduler config. [I have custom code based on `loadConfigFromFile`](https://github.com/kubernetes/autoscaler/pull/5708/files#diff-03046c2a86c17ed94a0d62f14449afb658d5a297113cbf894f3a1150858d0203R122-R141). It would help if we can have  `loadConfigFromFile` exported as `LoadConfigFromFile` (just like [`ValidateSchedulerConfiguration`](https://github.com/kubernetes/kubernetes/blob/746b88c6ff442185589640885228ef100811083a/pkg/scheduler/apis/config/validation/validation.go#L40)) so that programs which need to parse scheduler config can use the exported `LoadConfigFromFile` function directly instead of writing custom code.

### PR DESCRIPTION
**Pull request by vadasambar**
- so that users who import kube-scheduler libraries can use these functions to read kube scheduler config

#### What type of PR is this?

/kind feature (marked it as a feature but I think this is more of a `refactor`)

#### What this PR does / why we need it:
Check the issue: #119056
#### Which issue(s) this PR fixes:

Fixes #119056

#### Special notes for your reviewer:
NA
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by vadasambar** [comment]
I will squash the commits once the PR is approved.

File: cmd/kube-scheduler/app/options/configfile.go
Comment (aimuz) [inline cmd/kube-scheduler/app/options/configfile.go:33]: I think we should annotate the export function if we need it.

Comment (vadasambar) [reply cmd/kube-scheduler/app/options/configfile.go:33]: Ah, you are right. I didn't add a comment on top of the function. Thanks for noticing. I will add it :pray:

Comment (vadasambar) [reply cmd/kube-scheduler/app/options/configfile.go:34]: @aimuz I have added the comment.

Comment (aimuz) [reply cmd/kube-scheduler/app/options/configfile.go:34]: /lgtm

**Comment by vadasambar** [comment]
Both the test failures seem unrelated to the change in this PR. I wonder if the tests are flaky. 
image
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/119057/pull-kubernetes-integration/1678637778067263488
image
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/119057/pull-kubernetes-e2e-kind/1678637778117595136

**Comment by vadasambar** [comment]
Doesn't seem related to scheduler
image
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/119057/pull-kubernetes-unit/1678996872372424704

**Comment by vadasambar** [comment]
CI failure is not limited to this PR. CI for many PRs have failed due to what appears to be flaky CI: https://github.com/kubernetes/kubernetes/pulls

**Comment by vadasambar** [comment]
Tests passed but looks like lgtm label was removed. @aimuz sorry can I ask for lgtm label from you again :pray:

**Review by aimuz — APPROVED** [review]
/lgtm

**Comment by aimuz** [comment]
/assign alculquicondor 

for approval

**Comment by alculquicondor** [comment]
/hold
for thoughts from sig-autoscaling in issue.

**Comment by alculquicondor** [comment]
/hold cancel
based on https://github.com/kubernetes/autoscaler/pull/5708#issuecomment-1628479531

/approve

**Pull request merged**
